__tests__/dpop/storage.test.ts

```diff
@@ -0,0 +1,241 @@
+import { beforeEach, describe, expect } from '@jest/globals';
+import 'fake-indexeddb/auto';
+import { DpopStorage } from '../../src/dpop/storage';
+import {
+  TEST_CLIENT_ID,
+  TEST_DPOP_KEYPAIR,
+  TEST_DPOP_NONCE
+} from '../constants';
+
+describe('DpopStorage', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  let storage: DpopStorage;
+
+  beforeEach(async () => {
+    // start fresh
+    window.indexedDB = new IDBFactory();
+
+    storage = new DpopStorage(TEST_CLIENT_ID);
+  });
+
+  describe('createDbHandle()', () => {
+    const fakeNewVersion = Number.MAX_SAFE_INTEGER;
+
+    beforeEach(() => {
+      storage['getVersion'] = () => fakeNewVersion;
+    });
+
+    let output: IDBDatabase;
+
+    beforeEach(async () => {
+      output = await storage['createDbHandle']();
+    });
+
+    it('database is properly upgraded and created', () =>
+      expect(output.version).toBe(fakeNewVersion));
+  });
+
+  describe('getDbHandle()', () => {
+    const fakeDbHandle = {} as IDBDatabase;
+
+    beforeEach(() => {
+      storage = new DpopStorage(TEST_CLIENT_ID);
+      storage['createDbHandle'] = jest.fn().mockResolvedValue(fakeDbHandle);
+    });
+
+    describe('handle exists', () => {
+      beforeEach(() => {
+        storage['dbHandle'] = fakeDbHandle;
+      });
+
+      beforeEach(() => storage['getDbHandle']());
+
+      it('does not create handle', () =>
+        expect(storage['createDbHandle']).not.toHaveBeenCalled());
+    });
+
+    describe('otherwise', () => {
+      beforeEach(() => {
+        storage['dbHandle'] = undefined;
+      });
+
+      beforeEach(() => storage['getDbHandle']());
```

Comment:
Are multiple `beforeEach` calls needed? Seems to be a common pattern across the tests.